.buildkite/scripts/pr_build.sh

```diff
@@ -0,0 +1,70 @@
+#!/bin/bash
+
+set -exo pipefail
+
+# Script to run adhoc buildkite builds for the given branch. Useful for running
+# CI tests against branches that may not yet be ready for a PR. See
+# https://buildkite.com/docs/apis/rest-api#authentication for more info on
+# buildkite tokens. To take full advantage of this script your token will need
+# `read_builds` and `write_builds` scopes. The script expects a working
+# installation of jq.
+#
+# You'll also need a Github token with `public_repo` permissions.
+
+function usage() {
+  echo "build.sh \$PR_NUMBER"
+  exit 1
+}
+
+if [[ -z "$GITHUB_PR_TOKEN" ]]; then
+  echo "must set GITHUB_PR_TOKEN"
+  exit 1
+fi
+
+if [[ -z "$BUILDKITE_CLI_TOKEN" ]]; then
+  echo "must set BUILDKITE_CLI_TOKEN"
+  exit 1
+fi
+
+RESP=$(mktemp)
+
+function cleanup() {
+  rm -f "$RESP"
+}
+
+trap cleanup EXIT
+
+curl -H "Authorization: token $GITHUB_PR_TOKEN" -sSf -o "$RESP" "https://api.github.com/repos/m3db/m3/pulls/$1"
+
+function jq_field() {
+  <"$RESP" jq -r "$1"
```

Comment:
we should fail in `usage()` if `jq` isn't in `PATH`

Reply:
Hmm we kinda get that for free right now:

```
$ PATH=/usr/bin:/bin .buildkite/scripts/pr_build.sh 1759
...
.buildkite/scripts/pr_build.sh: line 40: jq: command not found
...
```

Reply:
_coincidentally_, it doesn't do anything bad before it fails due to `jq` not being installed. 😉 

agreed that in this case it's nbd, but generally we should do checks beforehand (also, a "jq must be installed and in `$PATH` to use this tool" message is less ambiguous than a "command not found" error).